src/main/java/org/rulez/demokracia/pdengine/VoteRegistry.java

```diff
@@ -16,12 +16,15 @@ public VoteRegistry(WebServiceContext wsContext) {
 	public String obtainBallot(String id, String adminKey) {
 		Vote vote = getVote(id);
 		
-		if(adminKey.equals("anon")) {
+		if(adminKey.equals(vote.adminKey))
```

Comment:
Well, the requirement was confusing. You can choose to leave the implementation as is, as this is not your mistake, and you are not obliged to read minds.
For the record I try to reword the intent of the two behaviours:
- adminkey can be either "user" or the adminkey of the vote
- if adminkey is "user", and the user have all the needed assurances, then the ballot is registered for the WS user, and at most one ballot can be issued for that user.
- if the adminkey is the vote's adminkey, then the ballot is registered to "admin"

Reply:
ok I will take a look later. :)

Reply:
So I pushed my changes and I observed that. You use user instead of anon. I mean I changed the anon checks but it is on the beharviors too.

For example: 
	@tested_feature("Manage votes")
	@tested_operation("Obtain ballot")
	@tested_behaviour("if adminKey is anon, the user should have all the neededAssurances")

Reply:
Btw it was a very good specificaton thanks :)

Reply:
Is there any future plan to explain more detailed the tasks/specification?

Reply:
Of course I have to update the architecture model. See #267